impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaConfig.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.testing.kafka.common;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import io.kroxylicious.testing.kafka.api.KafkaClusterConstraint;
+
+/**
+ * {@code @KafkaConfig} can be used to annotate a field in a test class or a
```

Comment:
I settled on the name `@KafkaConfig`.      The Kafka prefix is in sympathy with `KafkaCluster` type that is already part of the API. `@Config` would have been too general.

`KafkaConfig` does collude with `kafka.server.KafkaConfig` but given this is annotation I don't think it'll cause too much pain as the user probably won't be importing it anyway.   Why's that?  The constants defined by `kafka.server.KafkaConfig` aren't actually usable within the annotation declaration  This is because they are Scala constants which manifest in Java, as methods.

You'd like to write
```java
@KafkaConfig(name = kafka.server.KafkaConfig.DeleteTopicEnableProp(), value = "false")
```
but it won't compile.